### PR TITLE
Added: --strip cli option, ???

### PR DESCRIPTION
**Pull request by rmedaer**
The following commit implements a new feature: stripping the template directory.
When you create your project based on cookiecutter template, it's using the first "jinja formatted" directory as template directory and render it.
Even if you specify an output directory it will always create a sub-rendered-directory.

The strip option skip this first template directory and render the boilerplate in output directory.

Basically this allows to specify the output rendered directory:
```
cookiecutter --strip -o my-output-directory my-cookiecutter-template
```
or to render in working directory:
```
cookiecutter --strip -f -o . my-cookiecutter-template
```

The "strip" name is inspired from `tar` tool where `--strip-components` option _strip NUMBER leading components from file_.

### CODE REVIEW

**Comment by theodesp** [comment]
You need to fix the conflicting files. 👍

**Comment by kojiromike** [comment]
Anything I can do to help move this along?

**Comment by ekor15** [comment]
Same here - the usual flow for us is:
1. create a git repository
2. clone to local env
3. then run cookiecutter from that directory.

This will be very helpful

**Comment by insspb** [comment]
@rmedaer @ekor15 @kojiromike thank you for contribution and excuse us for long delay with answer. Some time project was in not maintainable state.

Is somebody still interested in this one?
I really do not fully understand how this works.
To be merged this pull request require some additional steps: 
1. Documentation with examples. 
2. Rebase to current master.
3. Some tests, that correctly shows how this works.
4. And this line should be commented, or option and variable renamed. `unrendered_dir if not strip else '',`

**Comment by alexlatchford** [comment]
Hey @insspb thanks for the update, I've actually reimplemented this fix as a part of `battenberg` (which is a fork of the work @rmedaer started to add automated upgrading to `cookiecutter` projects) and is working pretty well at Zillow. I'd be happy to close this out as I think we no longer need it.

**Comment by insspb** [comment]
@alexlatchford as from comments from this issue it seems that some other cc users needs that to, and if you find some time to implement this feature this will be great. If you check last commits to cc master branch you will see that we are implementing a lot of pull requests dept now, so no long delays on review here. I promise :)

**Comment by rmchale** [comment]
definitely interested in this one

**Comment by ofajardo** [comment]
definitely interested in this one

**Comment by simobasso** [comment]
Hey @ofajardo 👋 ,

As you can see, this pr is outdated, and the author is not interested anymore in bringing the code forward; we should wait for some contribution. Of course, if you have time to do it, feel free to open a new pr. It will be great!

I'll close this one.
